### ISSUE TITLE
[Accessibility]: Multiple H1 tags on Repository Page

### ISSUE THREAD

**Issue by gitstart-sourcegraph** [sourcegraph/sourcegraph#39291]
## Problem statement
There are multiple `<H1/>` elements: the Sourcegraph logo in the navigation bar, and the page title, which produces a warning in ARC Toolkit in the following pages::
1) [Repository detail page](https://k8s.sgdev.org/github.com/sourcegraph/sourcegraph)
2) [Repository Commits Page](https://k8s.sgdev.org/github.com/sourcegraph/sourcegraph/-/commits)
3) [ Repository Compare Page](https://k8s.sgdev.org/github.com/sourcegraph/sourcegraph/-/compare)

## Expected Behaviour
The page title should be an `<H2/>`, but retain the current style.
All other `<H2/>s `on the page should be downscaled to `<H3/>`, but retain their current styles.
No warnings are reported under ARC Toolkit's "Structure and Semantics" tests.

## Time estimate

- Pull requests with ~450 lines changed should take **X hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
- Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
